src/main/java/uk/gov/hmcts/darts/event/mapper/EventHandlerMapper.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.darts.event.mapper;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.darts.authorisation.api.AuthorisationApi;
+import uk.gov.hmcts.darts.common.entity.EventHandlerEntity;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
+import uk.gov.hmcts.darts.common.helper.CurrentTimeHelper;
+import uk.gov.hmcts.darts.event.model.EventMapping;
+import uk.gov.hmcts.darts.event.service.handler.DartsEventNullHandler;
+
+@Component
+@RequiredArgsConstructor
+public class EventHandlerMapper {
+
+    private final AuthorisationApi authorisationApi;
+    private final CurrentTimeHelper currentTimeHelper;
+
+    public EventHandlerEntity mapFromEventMappingAndMakeActive(EventMapping eventMapping) {
+        var eventHandlerEntity = new EventHandlerEntity();
+        eventHandlerEntity.setType(eventMapping.getType());
+        eventHandlerEntity.setSubType(eventMapping.getSubType());
+        eventHandlerEntity.setEventName(eventMapping.getName());
+        eventHandlerEntity.setHandler(getHandlerMappingOrNullHandler(eventMapping.getHandler()));
+        eventHandlerEntity.setIsReportingRestriction(eventMapping.getHasRestrictions());
+        eventHandlerEntity.setActive(true);
+        UserAccountEntity currentUser = authorisationApi.getCurrentUser();
+        eventHandlerEntity.setCreatedDateTime(currentTimeHelper.currentOffsetDateTime());
```

Comment:
I think this line can be removed - `EventHandlerEntity extends CreatedBaseEntity`, which uses the `@CreationTimestamp` annotation on the `createdDateTime` attribute. This means the timestamp _should_ auto-populate upon creation. Worth checking though!